docs: add detailed explanation for NormalModuleFactory hooks (#7468)
web-infra-dev/rspack: website/docs/en/api/plugin-api/normal-module-factory-hooks.mdx

# NormalModuleFactory

The `NormalModuleFactory` module is used by the [Compiler](/api/javascript-api/compiler) to generate modules. Starting with entry points, it resolves each request, parses the content to find further requests, and keeps crawling through files by resolving all and parsing any new files. At last stage, each dependency becomes a Module instance.

> The `NormalModuleFactory` class extends `Tapable` and provides the following lifecycle hooks. They can be tapped the same way as compiler hooks:

Reviewer: ```suggestion
`NormalModuleFactory` provides the following lifecycle hooks. These can be used just like `Compiler` hooks:
```

```js
NormalModuleFactory.hooks.someHook.tap(/* ... */);
```

> Besides `tap()`, you can also use `tapAsync()` and `tapPromise()`, depending on the type of hook.

Reviewer: ```suggestion
All hooks inherit from `Tapable`. In addition to `tap()`, you can also use `tapAsync()` and `tapPromise()`, depending on the type of the hook.
```
